native-image/microservices/micronaut-hello-rest-maven-layered/pom.xml

```diff
@@ -0,0 +1,254 @@
+              <artifactId>micronaut-http-validation</artifactId>
+              <version>${micronaut.core.version}</version>
+            </path>
+            <path>
+              <groupId>io.micronaut.serde</groupId>
+              <artifactId>micronaut-serde-processor</artifactId>
+              <version>${micronaut.serialization.version}</version>
+              <exclusions>
+                <exclusion>
+                  <groupId>io.micronaut</groupId>
+                  <artifactId>micronaut-inject</artifactId>
+                </exclusion>
+              </exclusions>
+            </path>
+          </annotationProcessorPaths>
+          <compilerArgs>
+            <arg>-Amicronaut.processing.group=example.micronaut</arg>
+            <arg>-Amicronaut.processing.module=micronaut-hello-rest-maven-layered</arg>
+          </compilerArgs>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>standalone</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.graalvm.buildtools</groupId>
+            <artifactId>native-maven-plugin</artifactId>
+            <version>0.10.3</version>
+            <configuration>
+              <imageName>standalone-app</imageName>
+              <mainClass>example.micronaut.Application</mainClass>
+              <buildArgs>
+                <buildArg>--initialize-at-run-time=io.netty</buildArg>
+              </buildArgs>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>base-layer</id>
+      <build>
+        <directory>${project.basedir}/base-layer-target</directory>
+        <plugins>
+          <plugin>
+            <groupId>org.graalvm.buildtools</groupId>
+            <artifactId>native-maven-plugin</artifactId>
+            <version>0.10.3</version>
+            <configuration>
+              <imageName>libmicronautbaselayer</imageName>
+              <mainClass>.</mainClass>
+              <buildArgs>
+                <buildArg>-H:+UnlockExperimentalVMOptions</buildArg>
+                <buildArg>-H:LayerCreate=base-layer.nil,module=java.base,package=io.micronaut.*,package=io.netty.*,package=jakarta.*,package=com.fasterxml.jackson.*,package=org.slf4j.*,package=reactor.*,package=org.reactivestreams.*</buildArg>
+                <buildArg>-H:ApplicationLayerOnlySingletons=io.micronaut.core.io.service.ServiceScanner$StaticServiceDefinitions</buildArg>
+                <buildArg>-H:ApplicationLayerInitializedClasses=io.micronaut.inject.annotation.AnnotationMetadataSupport</buildArg>
+                <buildArg>-H:ApplicationLayerInitializedClasses=io.micronaut.core.io.service.MicronautMetaServiceLoaderUtils</buildArg>
+                <buildArg>-H:-UnlockExperimentalVMOptions</buildArg>
+                <buildArg>--initialize-at-run-time=io.netty</buildArg>
+                <buildArg>-J-Xmx6g</buildArg>
```

Comment:
Why set max heap here? to demonstrate this builds within 6GB? add a comment? Also, this flag is not in the README.md, not sure that's intentional?

Reply:
In this case, you don't need to set the flag at all because GHA sets `CI=true` and then the build process switches into dedicated mode.

Reply:
I removed all Xmx, in case it fails in GHA we can update it.